### PR TITLE
Fix memory leak with exit files

### PR DESCRIPTION
**Pull request by mheon**
There's an excellent summary of the leak itself in the message of the first commit; suffice to say, we can't leave exit files around in tmpfs, it's causing a memory leak.

Instead, just use the events system to retrieve exit codes.

### CODE REVIEW

**Comment by mheon** [comment]
Many thanks to Andrea Arcangeli for tracking the leak here down.

**Comment by mheon** [comment]
One test failing - `podman run --http-proxy`. Investigating.

**Comment by mheon** [comment]
It's an exit code with a `--rm` container, which is directly affected by this patch. Not clear why it's broken - works fine locally.

**Comment by rhatdan** [comment]
LGTM

**Comment by mheon** [comment]
Test failures look legitimate, will take a look later today

**Comment by mheon** [comment]
Problem: podman-in-podman tests. We're using events logging to retrieve the exit code - but there's no journald available for podman-in-podman, so no events. Evidently the result is that Podman times out?

**Comment by mheon** [comment]
I'm going to add a flag to manually specify events backend and force use of an events file

**Comment by mheon** [comment]
Updated. Tests should now all use the file backend for events.

**Comment by mheon** [comment]
`[+3876s] Error: error creating libpod runtime: error configuring CNI network plugin: could not create new watcher too many open files`

Eeek? I think we might be leaking FDs for events

**Comment by mheon** [comment]
Hm. I bet it's image caching/uncaching again... we're leaking a single lockfile per libpod runtime created.

**Comment by mheon** [comment]
I think we need to expose the /dev/null events backend and force it for uncaching/caching calls

**Comment by mheon** [comment]
Pushed a fresh version that exposes the null eventer and uses it for uncaching.

**Comment by TomSweeneyRedHat** [comment]
LGTM if you can chase down the test issues.

**Comment by mheon** [comment]
Lots of timeouts now. It looks like it's probably from reading events?

**Comment by mheon** [comment]
Alright, given the problems here, I'm probably going to split this into two PRs.

The first one will carry only the memory leak fix. The second will carry everything else, getting exit codes with `--rm` containers working again.

**Comment by mheon** [comment]
Hm. So we are definitely hitting the right code paths here - you can see it trying to get into events, it just can't find the `died` event.

I don't think this is a timing thing - if we're in this code, the container has already been removed, and remove should guarantee a Died event being recorded as part of it. Going to investigate further.

**Comment by mheon** [comment]
Hm. `podman events` is exiting with no output, but exit code 1.

**Comment by mheon** [comment]
Alright. There's a problem where at least some events are not being displayed (probably why events is throwing a nonzero exit code), but the core issue is that events only sees a single event - a refresh. Part of me wonders if we're seeing something happening that wipes the libpod directory before the last command is finished executing.

**Comment by mheon** [comment]
Finally green
@rhatdan @baude @TomSweeneyRedHat @vrothberg @giuseppe PTAL

File: docs/libpod.conf.5.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ libpod to manage containers.
   a slirp4netns network.  If "" is used then the binary is looked up using the $PATH environment variable.
 
 **events_logger**=""
-  Default method to use when logging events. Valid values are "journald" and "file".
+  Default method to use when logging events. Valid values are "file", "journald", and "null".
```
Comment (rhatdan) [inline docs/libpod.conf.5.md]: What does "null" mean?  If it means "none" should we use that?  Seems like a more Human name for it.

File: docs/podman.1.md [hunk 2]
```diff
@@ -36,6 +36,10 @@ Note: CGroup manager is not supported in rootless mode when using CGroups Versio
 
 Path to where the cpu performance results should be written
 
+**--events-logger**=**type**
```
Comment (edsantiago) [inline docs/podman.1.md]: Nit: the convention everywhere else seems to be to use italics (one asterisk) for the arg value.

File: docs/libpod.conf.5.md (hunk 1 above)
Comment (rhatdan) [reply docs/libpod.conf.5.md]: After reading, I think we should present to the user NONE, Null is a programmer term and does not carry the same meaning as None.

Comment (mheon) [reply docs/libpod.conf.5.md]: "null" as in "forward to /dev/null" - I'm open to changing this one, though. None also works.

**Comment by mheon** [comment]
Comments addressed - "null" renamed to "none" on the command line

**Comment by rhatdan** [comment]
LGTM
@baude @vrothberg @giuseppe @TomSweeneyRedHat @QiWang19 @ashley-cui PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Comment by mheon** [comment]
It's green, let's get this landed

**Pull request merged**